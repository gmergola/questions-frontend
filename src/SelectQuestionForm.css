.Form-container {
  margin-top: 10%;
  margin-bottom: 100%;
}

.Question-select {
  width: 200px;
  height: 30px;
  border-radius: 20px;
  border: none;
}

.SelectAnswerForm-option {
  color: green;
}

.choose-label {
  font-size: 30px;
  margin-left: 300px;
}

.choose-button {
  margin-top: 30px;
  border-radius: 20px;
  width: 200px;
  height: 30px;
  border: none;
  margin-left: 590px;
}

.choose-button:hover {
  opacity: 0.5;
}

@media only screen and (max-width: 900px){
  .Question-select {
    width: 120px;
  }

  .choose-label {
    font-size: 20px;
    margin-left: 50px;
  }

  .choose-button {
    margin-left: 150px;
    width: 150px;
  }

}