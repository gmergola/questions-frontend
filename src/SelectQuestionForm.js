import React, { useState } from 'react';
import FamilyFeudApi from './apiHelpers';
import useApi from './hooks/useApi';
import { useHistory } from "react-router-dom";
import './SelectQuestionForm.css';

/**SelectQuestionForm: a component to allow the user to select a question to answer */
function SelectQuestionForm() {
  const [loading, data] = useApi(FamilyFeudApi.getQuestions);
  const [questionIdx, setQuestionIdx] = useState(0);
  const history = useHistory();

  const handleChange = evt => {
    setQuestionIdx(evt.target.value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    history.push(`/questions/${data.questions[questionIdx].question_main}`);
  }

  return (
    loading ? "Loading...":
      <div className="Form-container">
        <form onSubmit={handleSubmit}>
          <label className="choose-label" htmlFor="questions">Choose a Question: </label>
          <select name="questions" className="Question-select" onChange={handleChange}>
            {data.questions.map((q, idx) => (
              <option key={q.question_main} value={idx}>
                {q.question_main}
              </option>
            ))}
          </select>
          <div>
            <button className="choose-button"><b>Choose Question</b></button>
          </div>
        </form>
      </div>
  );
}

export default SelectQuestionForm;