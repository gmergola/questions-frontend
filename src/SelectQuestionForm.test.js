import React from "react";
import { render } from "@testing-library/react";
import SelectQuestionForm from "./SelectQuestionForm";

describe("renders sucessfully", () => {
  it("renders without crashing", () => {
    render(<SelectQuestionForm />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<SelectQuestionForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});