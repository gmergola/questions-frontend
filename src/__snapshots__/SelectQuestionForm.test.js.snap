// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sucessfully matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form-container"
  >
    <div
      class="spinner"
    />
  </div>
</DocumentFragment>
`;
